core/server/common/src/main/java/alluxio/master/journal/raft/RaftJournalWriter.java

```diff
@@ -62,25 +64,34 @@ public RaftJournalWriter(long nextSequenceNumberToWrite,
     mNextSequenceNumberToWrite = new AtomicLong(nextSequenceNumberToWrite);
     mLastSubmittedSequenceNumber = new AtomicLong(-1);
     mLastCommittedSequenceNumber = new AtomicLong(-1);
+    mCurrentJournalEntrySize = new AtomicLong(0);
     mClient = client;
     mClosed = false;
     mWriteTimeoutMs =
         ServerConfiguration.getMs(PropertyKey.MASTER_EMBEDDED_JOURNAL_WRITE_TIMEOUT);
+    // journal entry size max is the hard limit set by underlying ratis
+    // use a smaller value to guarantee we don't pass the hard limit
+    mFlushBatchBytes =
+        ServerConfiguration.getBytes(PropertyKey.MASTER_EMBEDDED_JOURNAL_ENTRY_SIZE_MAX) / 3;
```

Comment:
`max / 3` is basically a heuristic, right? What would happen if this was larger, like `max / 2`? What are the downsides?

Reply:
Both max/3 and max/2 have no obvious difference in my new flush() in write().
If an entry is bigger than max, i will directly log an error. both max/3 and max/2 cannot avoid the failure in this case.

I set max / 3 mainly because we currently give the entry size max 10MB which is a relatively big number (the default entry size max is 3MB). 3MB journal entry size is big enough for one flush().